packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -609,6 +656,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return idx;
       }
 
+      this.tsParseMethodSignatureKind(node);
       this.parsePropertyName(node, /* isPrivateNameAllowed */ false);
```

Comment:
Instead of using `lookahead` (which is expensive) in `tsParseMethodSignatureKind`, can we:
- Call `this.parsePropertyName`
- If not computed, not private, does not have type parameters, and the identifier name is `get` or `set`: 
  - If `tsNextTokenCanFollowModifier`, then we are parsing an accessor and we call `this.parsePropertyName` again
  - Otherwise, it's a method/property named `get` or `set`
- Otherwise, it's a method/property

Reply:
Yeah we parse class accessors using the same strategy mentioned above.

Reply:
What the AST babel should return for below code?

```ts
interface Foo {
    set foo;
}
```

In current behaviour, two properties named `set` and `foo` are returned (with recoverable error `Missing semicolon`). If we should keep it, we should detect if the property is a method signature after first `this.parsePropertyName` call. But then the token is the name of the property, so I don't know how to do that without lookahead...

Reply:
It's ok to throw "Expected (":
- It's what we already do for
  ```js
  class Foo {
    set foo
  }
  ```
- It's more likely that someone didn't finish writing a setter, than putting two properties on the same line without `;` and that the name of the first one was exactly `set`.